clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -12,31 +12,59 @@
  */
 
+import { IZoweLogger, MessageSeverityEnum } from "@zowe/zowe-explorer-api/lib/logger";
+import { ZoweVsCodeExtension } from "@zowe/zowe-explorer-api/lib/vscode";
 import * as fs from "fs";
+import { homedir } from "os";
+import * as Path from "path";
 import * as vscode from "vscode";
 import {
-    CONN_REFUSED_ERROR_MSG, DOWNLOAD_QUEUE_LOCKED_ERROR_MSG,
-    DSN_NOT_FOUND_ERROR_MSG,
-    DSN_PLACEHOLDER,
+    DOWNLOAD_QUEUE_LOCKED_ERROR_MSG,
     INVALID_CREDENTIALS_ERROR_MSG,
-    NO_PASSWORD_ERROR_MSG,
     PROCESS_DOWNLOAD_ERROR_MSG,
-    PROFILE_NAME_PLACEHOLDER, UNLOCK_DOWNLOAD_QUEUE_MSG,
+    PROFILE_NAME_PLACEHOLDER, SETTINGS_CPY_SECTION, UNLOCK_DOWNLOAD_QUEUE_MSG,
 } from "../../constants";
-import {CopybookProfile, DownloadQueue} from "./DownloadQueue";
-import {ProfileService} from "../ProfileService";
-import {TelemetryService} from "../reporter/TelemetryService";
-import {ZoweApi} from "../ZoweApi";
-import {Type, ZoweError} from "../ZoweError";
-import {checkWorkspace, CopybooksPathGenerator, createCopybookPath, createDatasetPath} from "./CopybooksPathGenerator";
+import { TelemetryService } from "../reporter/TelemetryService";
+import { ProfileUtils } from "../util/ProfileUtils";
+// tslint:disable-next-line:max-line-length
+import { checkWorkspace, CopybooksPathGenerator, createCopybookPath, createDatasetPath } from "./CopybooksPathGenerator";
+import { CopybookProfile, DownloadQueue } from "./DownloadQueue";
 
+const experimentTag = "experiment-tag";
 export class CopybookDownloadService implements vscode.Disposable {
+    private static async showQueueLockedDialog(profileName: string): Promise<boolean> {
+        const action = await vscode.window.showErrorMessage(
+            DOWNLOAD_QUEUE_LOCKED_ERROR_MSG.replace(PROFILE_NAME_PLACEHOLDER, profileName),
+            UNLOCK_DOWNLOAD_QUEUE_MSG);
+        return action === UNLOCK_DOWNLOAD_QUEUE_MSG;
+    }
+
+    private static createErrorMessageForCopybooks(datasets: Set<string>) {
+        CopybookDownloadService.processDownloadError(PROCESS_DOWNLOAD_ERROR_MSG + Array.from(datasets));
+    }
+
+    private static processDownloadError(title: string) {
+        const actionSettings = "Change settings";
+        vscode.window.showErrorMessage(title, actionSettings).then(action => {
+            if (action === actionSettings) {
+                vscode.commands.executeCommand("cobol-lsp.cpy-manager.goto-settings");
+            }
+        });
+    }
+
+    private static needsUserNotification(queue: CopybookProfile[]): boolean {
+        return queue.find(profile => !profile.quiet) !== undefined;
+    }
+
+    private static isInvalidCredentialsError(error: any) {
+        return error?.mDetails?.errorCode === 401;
+    }
+
     private queue: DownloadQueue = new DownloadQueue();
     private lockedProfile: Set<string> = new Set();
+    private CobolLSzoweLogger = new IZoweLogger("ZOWE Explorer", `${homedir()}/LSPCobol/logs`);
```

Comment:
Does it have any settings for the log level?

Reply:
Good point. I will look into the docs. Thanks

Reply:
pretty weird. I do see a logging config but it's part of zoweAPI project.

config - https://github.com/zowe/vscode-extension-for-zowe/blob/master/packages/zowe-explorer-api/src/log4jsconfig.json

logger - https://github.com/zowe/vscode-extension-for-zowe/blob/master/packages/zowe-explorer-api/src/logger/IZoweLogger.ts

It's set to DEBUG.

Reply:
I see that you use it only for showing Zowe errors. But why? We can log things ourselves and show errors to the user as well.
I mean that instead of:
```
ZoweVsCodeExtension.showVsCodeMessage(...
```
use something like:
```
vscode.window.showErrorMessage(
```

Reply:
BTW, are you sure that the name must start with the capital letter?

Reply:
Yes. I agree. But I was thought to use their service to show messages on vscode as well as to log it, in their log format way. I kind of don't have any particular inclination for it, so I may change it if the team feels the other way.

For the name, whatever name we give, is shown in the message window, something like below screen shot
![image](https://user-images.githubusercontent.com/69206564/145029584-605d1e3e-3f55-4e75-b693-74a876759775.png)

Reply:
removed the zowe logger